packages/common/src/i18n/locale_data_api.ts

```
@@ -399,11 +399,12 @@ export class LocaleDataApi {
  * Returns the first value that is defined in an array, going backwards
  */
 function firstDefined<T>(data: T[], index: number): T {
-  let res;
-  for (let i = index; i > -1 && typeof res === 'undefined'; i--) {
-    res = data[i];
+  for (let i = index; i > -1; i--) {
+    if (typeof data[i] !== 'undefined') {
+      return data[i];
+    }
   }
-  return res !;
+  return data[index];
```

we could also throw an error instead, since this use case is not supposed to happen